os/kernel/log_dump/log_dump.c

```diff
@@ -88,15 +102,28 @@ static struct log_dump_chunk_s *read_node;
  * Public Functions
  ****************************************************************************/
 
+mqd_t log_dump_get_mqfd(void)
+{
+	return g_logdump_mq_fd;
+}
+
 int log_dump_init(void)
 {
 	sq_init(&log_dump_chunks);
 	/* no chunks are present so we allocate new chunk */
 	struct log_dump_chunk_s *node = (struct log_dump_chunk_s *)kmm_malloc(LOG_CHUNK_SIZE);
 
 	if (node == NULL) {
+		lldbg("memory allocation failure\n");
 		return LOG_DUMP_MEM_FAIL;
 	}
+	uncomp_array = (signed char *)kmm_malloc(CONFIG_LOG_DUMP_CHUNK_SIZE);
+	if (uncomp_array == NULL) {
+		lldbg("memory allocation failure\n");
+		kmm_free(node);
+		return LOG_DUMP_MEM_FAIL;
+	}
+	node->comp_head = 0;		/* since first node */
 
 	curbytes = 0;
 	current_ptr = 0;
```

Comment:
It's much hard to know the purpose of each. Could you rename?

Reply:
renamed. Added description at declaration